File: weblayer/public/java/org/chromium/weblayer/ErrorPage.java
// Copyright 2020 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.weblayer;

import androidx.annotation.NonNull;

/**
 * ErrorPage contains the html to show when an error is encountered.
 */
class ErrorPage {
    public final String htmlContent;

    /**
     * Creates an ErrorPage.
     *
     * @param htmlContent The html to show.
     *
     */
    public ErrorPage(@NonNull String htmlContent) {
        this.htmlContent = htmlContent;
    }
}

File: chrome/browser/resources/contextual_tasks/error_page.ts
// Copyright 2025 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import {CrLitElement} from 'chrome://resources/lit/v3_0/lit.rollup.js';

import {getCss} from './error_page.css.js';
import {getHtml} from './error_page.html.js';

export class ContextualTasksErrorPageElement extends CrLitElement {
  static get is() {
    return 'error-page';
  }

  static override get styles() {
    return getCss();
  }

  static override get properties() {
    return {};
  }

  override render() {
    return getHtml.bind(this)();
  }

  override connectedCallback() {
    super.connectedCallback();
  }

  override disconnectedCallback() {
    super.disconnectedCallback();
  }
}

declare global {
  interface HTMLElementTagNameMap {
    'error-page': ContextualTasksErrorPageElement;
  }
}

customElements.define(
    ContextualTasksErrorPageElement.is, ContextualTasksErrorPageElement);
